src/signatures.jl

```diff
@@ -377,6 +389,9 @@ function replacename!(args::Vector{Any}, pr)
             args[i] = QuoteNode(newname)
         elseif isa(a, Vector{Any})
             replacename!(a, pr)
+        elseif isa(a, Core.ReturnNode) && isdefined(a, :val) && a.val isa Expr
+            # there is something like `ReturnNode(Expr(:method, Symbol(...)))`
+            replacename!(a.val::Expr, pr)
```

Comment:
This is necessary especially because `CodeInfo` wrapped in `:thunk` expression contains `ReturnNode(Expr(:method, gensym))`.